riscv-c-api.md

```diff
@@ -164,11 +164,26 @@ For example:
 
 ## Function Attributes
 
-* `__attribute__((naked))`
-* `__attribute__((interrupt))`
-* `__attribute__((interrupt("user")))`
-* `__attribute__((interrupt("supervisor")))`
-* `__attribute__((interrupt("machine")))`
+### `__attribute__((naked))`
+
+The compiler won't generate the prologue/epilogue for those functions with
+`naked` attributes, that usually used when you want to write a function with
+an inline assembly body.
```

Comment:
We're getting to the point where we risk just reproducing documentation that should really belong in the compilers docs, but a key difference here is that non-asm statements are not allowed in naked functions in Clang. Unsure if it's worth mentioning this or not.

Reply:
We should not duplicate compiler-specifics here. They might change, and we end up with outdated information here.

Still, the hint is helpful. What about a more generic note like "Note: Be aware that compilers might have further restrictions on naked functions. Please consult your compiler's manual for more information."

Reply:
Actually I am prepare to propose `target` attribute, but I found function attribute section already there but almost empty body, so I try to fill this empty before I post `target` attribute part.

For `naked` and `interrupt`, that might seem like reproducing doc. but `target` won't be the same case I believe :)